core/modules/replica/protocol.proto

```diff
@@ -168,25 +170,71 @@ message ProtocolRequestEcho {
     optional uint64 delay = 3 [default = 0];
 }
 
+// This message is used in the table schema definitions
+message ProtocolRequestSqlColumn {
+    required string name = 1;
+    required string type = 2;
+}
+
 // This request is always queued.
-// This message is sent once after the header.
+// This message is sent once after the header for the 'SQL' request types
 message ProtocolRequestSql {
 
     required int32 priority = 1;
 
-    /// The query to be executed
-    required string query = 2;
-
-    /// The name of a database user authorized in a context of the query
-    required string user = 3;
-
-    /// Database password for the user account
-    required string password = 4;
+    enum Type {
+        QUERY = 0;
+        CREATE_DATABASE = 1;
+        DROP_DATABASE = 2;
+        ENABLE_DATABASE = 3;    // in Qserv
+        DISABLE_DATABASE = 4;   // in Qserv
+        GRANT_ACCESS = 5;
+        CREATE_TABLE = 6;
+        DROP_TABLE = 7;
+        REMOVE_TABLE_PARTITIONING = 8;
+        DROP_TABLE_PARTITION = 9;
+    }
+    required Type type = 2;
```

Comment:
You are not supposed to reuse field numbers.

Reply:
Fixed